### PR TITLE
20802 static documents with file name

### PR DESCRIPTION
**Pull request by vysakh-menon-aot**
*Issue #:* /bcgov/entity#20802

*Description of changes:*
 - new staticDocuments object with name and url
```
 {
    "documents": {
        "certificate": "http://localhost:5000/api/v2/businesses/C9900731/filings/149329/documents/certificate",
        "legalFilings": [
            {
                "continuationIn": "http://localhost:5000/api/v2/businesses/C9900731/filings/149329/documents/continuationIn"
            }
        ],
        "noticeOfArticles": "http://localhost:5000/api/v2/businesses/C9900731/filings/149329/documents/noticeOfArticles",
        "receipt": "http://localhost:5000/api/v2/businesses/C9900731/filings/149329/documents/receipt",
        "staticDocuments": [
            {
                "name": "Director Affidavit",
                "url": "http://localhost:5000/api/v2/businesses/C9900731/filings/149329/documents/static/8c0c4a4f-41cf-4934-a27c-81fca942d05e.pdf"
            },
            {
                "name": "BC0923795 Annual Report (2024) - 2024-05-24.pdf",
                "url": "http://localhost:5000/api/v2/businesses/C9900731/filings/149329/documents/static/908d6a5e-a957-49c5-b671-7ed75b6d1d9d.pdf"
            }
        ]
    }
}
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: legal-api/src/legal_api/core/filing.py [hunk 1]
```diff
@@ -557,4 +557,8 @@ def get_document_list(business,  # pylint: disable=too-many-locals disable=too-m
                 for doc in additional:
                     documents['documents'][doc] = f'{base_url}{doc_url}/{doc}'
 
+                if has_roles(jwt, [UserRoles.staff]):
+                    if static_docs := FilingMeta.get_static_documents(filing.storage, f'{base_url}{doc_url}/static'):
+                        documents['documents']['staticDocuments'] = static_docs
```
Comment (severinbeauvais) [inline legal-api/src/legal_api/core/filing.py:562]: Only staff receive these 👍

File: legal-api/src/legal_api/core/meta/filing.py [hunk 2]
```diff
@@ -619,6 +620,26 @@ def alter_outputs_special_resolution(filing, outputs):
...
 
+    @staticmethod
+    def get_static_documents(filing, url_prefix):
+        """Get static documents."""
+        outputs = []
+        if filing.filing_type == 'continuationIn':
+            continuation_in = filing.meta_data.get('continuationIn', {})
+            if file_key := continuation_in.get('affidavitFileKey'):
+                outputs.append({
+                    'name': 'Director Affidavit',
```
Comment (severinbeauvais) [inline legal-api/src/legal_api/core/meta/filing.py:631]: "Director" or "Director's" ? I've seen both. Please check with Andy.

File: legal-api/src/legal_api/resources/v2/business/business_filings/business_documents.py [hunk 3]
```diff
@@ -70,23 +71,32 @@ def get_documents(identifier: str, filing_id: int, legal_filing_name: str = None
                                       **{'filing_id': filing_id, 'identifier': identifier})
         ), HTTPStatus.NOT_FOUND
 
-    if not legal_filing_name:
+    if not legal_filing_name and not file_key:
         if identifier.startswith('T') and filing.status == Filing.Status.COMPLETED:
-            return {}, HTTPStatus.NOT_FOUND
+            return {'documents': {}}, HTTPStatus.OK
```
Comment (severinbeauvais) [inline legal-api/src/legal_api/resources/v2/business/business_filings/business_documents.py:76]: Yay, empty documents object for T filings/businesses! Thanks :)

File: legal-api/src/legal_api/resources/v2/business/business_filings/business_documents.py [hunk 4]
```diff
@@ -70,23 +71,32 @@ def get_documents(identifier: str, filing_id: int, legal_filing_name: str = None
...
-        return get_pdf(filing.storage, legal_filing_name)
+    if 'application/pdf' in request.accept_mimetypes:
+        if legal_filing_name:
+            if legal_filing_name.lower().startswith('receipt'):
+                return _get_receipt(business, filing, jwt.get_token_auth_header())
+
+            return get_pdf(filing.storage, legal_filing_name)
+        elif file_key and (document := Document.find_by_file_key(file_key)):
+            if document.filing_id == filing.id:  # make sure the file belongs to this filing
+                response = MinioService.get_file(document.file_key)
```
Comment (severinbeauvais) [inline legal-api/src/legal_api/resources/v2/business/business_filings/business_documents.py:87]: How long is the download link good for?

**Review by severinbeauvais — APPROVED** [review]

**Review by ketaki-deodhar — APPROVED** [review]
looks good

File: legal-api/src/legal_api/resources/v2/business/business_filings/business_documents.py (hunk 4 above)
Comment (vysakh-menon-aot) [reply legal-api/src/legal_api/resources/v2/business/business_filings/business_documents.py:87]: this function returns the file not the link (so no expiry)

**Review by argush3 — APPROVED** [review]

File: legal-api/src/legal_api/core/meta/filing.py (hunk 2 above)
Comment (severinbeauvais) [reply legal-api/src/legal_api/core/meta/filing.py:631]: This shows "Director Affidavit":

image
https://www.figma.com/design/wVF8EcUCUR20YyqA0wHVJp/Continuation-In-V5?node-id=56-5091&t=9KjFzGFK37HPQIKH-0

Comment (severinbeauvais) [reply legal-api/src/legal_api/core/meta/filing.py:631]: This shows a few options:

image
https://www.figma.com/design/wVF8EcUCUR20YyqA0wHVJp/Continuation-In-V5?node-id=56-2768&t=9KjFzGFK37HPQIKH-0

Comment (vysakh-menon-aot) [reply legal-api/src/legal_api/core/meta/filing.py:631]: "Director Affidavit". Confirmed with Andy

**Pull request merged**